src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,60 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  constructor(name, health = 100) {
+    super(name, health);
+
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    return (this.hidden = true);
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (animal instanceof Carnivore) {
+      return;
+    }
+
+    if (animal.hidden === true) {
+      return;
+    }
+
+    animal.health -= 50;
+
+    if (animal.health === 0) {
```

Comment:
[CODE PERFORMANCE] - The condition `if (animal.health === 0)` is not sufficient to handle cases where health drops below zero. It should be `if (animal.health <= 0)` to catch all scenarios where the animal's health is zero or negative.